io/iommu/iommu_tests.py

```diff
@@ -0,0 +1,373 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2022 Advanced Micro Devices, Inc.
+# Author: Dheeraj Kumar Srivastava <dheerajkumar.srivastava@amd.com>
+
+"""
+Various combination of unbind, bind, change iommu group domain type, reset
+and rescan is used to form sub-tests that test and exercise iommu code.
+"""
+
+import os
+from avocado import Test
+from avocado.utils import process
+from avocado.utils import pci, genio
+from avocado.utils.software_manager.manager import SoftwareManager
+from avocado.utils import wait
+
+
+def change_domain_check(dom, pci_addr, def_dom):
+    """
+    Check if the domain changed successfully to "dom" for "pci_addr"
+
+    :param dom: domain type
+    :param def_dom: default domain of pci device(pci_addr)
+    :param pci_addr: pci device
+    return: bool
+    """
+    output = genio.read_one_line(
+            f"/sys/bus/pci/devices/{pci_addr}/iommu_group/type")
+    out = output.rsplit(None, 1)[-1]
+    if (dom not in ("auto", out)) or (dom == "auto" and out != def_dom):
+        return False
+    return True
+
+
+def reset_check(pci_addr):
+    """
+    Check if reset for "pci_addr" is successful
+
+    :param pci_addr: pci device
+    return: bool
+    """
+    cmd = f"lspci -vvs {pci_addr}"
+    output = process.run(cmd, ignore_status=True, shell=True).stdout_text
+    if output != "":
+        return False
+    return True
+
```

Comment:
why not push these in the avocado library itself , can be have in this utility 
https://github.com/avocado-framework/avocado/blob/master/avocado/utils/pci.py

Reply:
Because of the issue https://github.com/avocado-framework/avocado/issues/5417, we are not able to use latest upstream avocado.
so, please merge this in tests now, we will take this as enhancement to avocado once that issue is fixed.

Reply:
ah sure missed from my TODO, will have some thing on  YAML issue soon, but still, i say please have part of the utils itself

Reply:
Yeah sure, once the issue is resolved will be moving the parts to avocado utils.

Reply:
@PraveenPenguin Can u please the parts that we can have in avocado for reuse?

Reply:
Seems, reset_check() and rescan_check() are two candidate to pci.py.
Let me know other functions to move to avocado/utils/pci.py.